cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/worker/TaskSystemAppContext.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.service.worker;
+
+import io.cdap.cdap.api.Admin;
+import io.cdap.cdap.api.ServiceDiscoverer;
+import io.cdap.cdap.api.artifact.ArtifactManager;
+import io.cdap.cdap.api.macro.InvalidMacroException;
+import io.cdap.cdap.api.macro.MacroEvaluator;
+import io.cdap.cdap.api.macro.MacroParserOptions;
+import io.cdap.cdap.api.plugin.PluginConfigurer;
+import io.cdap.cdap.api.security.store.SecureStore;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * System App context for a remote task
+ */
+public interface TaskSystemAppContext {
+
+  Admin getAdmin();
+
+  Map<String, String> getPreferencesForNamespace(String namespace, boolean resolved) throws IOException;
+
+  PluginConfigurer createPluginConfigurer(String namespace) throws IOException;
+
+  Map<String, String> evaluateMacros(String namespace, Map<String, String> macros,
+                                     MacroEvaluator evaluator,
+                                     MacroParserOptions options) throws InvalidMacroException;
+
+  ServiceDiscoverer getServiceDiscoverer();
```

Comment:
It looks like implementation implements `ServiceDiscoverer,
  SecureStore`. Could we extend those interfaces here?

Reply:
Done.